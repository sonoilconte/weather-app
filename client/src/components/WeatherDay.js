import React from 'react';

const dateOptions = { weekday: 'short', year: 'numeric', month: 'long', day: 'numeric' };

class WeatherDay extends React.Component {
	render() {
		const {
			date,
			high,
			low,
		} = this.props.day;
		const formattedDate = new Date(date).toLocaleDateString('en-US', dateOptions);
		return (
			<div className="card">
				<div className="card-body">
					<div className="card-title">{formattedDate}</div>
					<div className="card-text">High: {high}&deg;F</div>
					<div className="card-text">Low: {low}&deg;F</div>
				</div>
			</div>
		);
	}
}

export default WeatherDay;