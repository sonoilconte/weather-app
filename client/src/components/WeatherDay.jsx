/* eslint-disable react/jsx-one-expression-per-line */
import React from 'react';
import PropTypes from 'prop-types';

const dateOptions = {
    weekday: 'short',
    year: 'numeric',
    month: 'long',
    day: 'numeric',
};

function WeatherDay({
    date,
    high,
    low,
}) {
    const formattedDate = new Date(date).toLocaleDateString('en-US', dateOptions);
    return (
        <div className="card">
            <div className="card-body">
                <div className="card-title">{formattedDate}</div>
                <div className="card-text">High: {high}&deg;F</div>
                <div className="card-text">Low: {low}&deg;F</div>
            </div>
        </div>
    );
}

WeatherDay.propTypes = {
    date: PropTypes.string.isRequired,
    high: PropTypes.number.isRequired,
    low: PropTypes.number.isRequired,
};

export default WeatherDay;
