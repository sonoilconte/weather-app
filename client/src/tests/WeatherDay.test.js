import renderer from 'react-test-renderer';

import WeatherDay from '../components/WeatherDay';

it('renders the WeatherDay component correctly', () => {
	const tree = renderer.create(<WeatherDay day={{ date: '2022-10-15', high: 75.3, low: 56.8 }} />)
		.toJSON();
	expect(tree).toMatchSnapshot();
})
