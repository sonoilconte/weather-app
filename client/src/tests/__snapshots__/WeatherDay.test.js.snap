// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the WeatherDay component correctly 1`] = `
<div
  className="card"
>
  <div
    className="card-body"
  >
    <div
      className="card-title"
    >
      Fri, October 14, 2022
    </div>
    <div
      className="card-text"
    >
      High: 
      75.3
      °F
    </div>
    <div
      className="card-text"
    >
      Low: 
      56.8
      °F
    </div>
  </div>
</div>
`;
